src/library_sockfs.js

```diff
@@ -194,6 +194,13 @@ mergeInto(LibraryManager.library, {
             // The node ws library API for specifying optional subprotocol is slightly different than the browser's.
             var opts = ENVIRONMENT_IS_NODE ? {'protocol': subProtocols.toString()} : subProtocols;
 
+            // some webservers (azure) does not support subprotocol header
+
+            if (runtimeConfig && null === Module['websocket']['subprotocol']) {
+                subProtocols = 'null';
+                opts = undefined;
```

Comment:
why is `opts` also changed?

Reply:
Because it's passed as argument in constructor
```
ws = new WebSocketConstructor(url, opts);
```

opts contains only subprotocol field. If subprotocol is null then opts also is empty (undefined). I set protocols to 'null' to avoid exception on next line

```
              Module.print('connect: ' + url + ', ' + subProtocols.toString());
```

I need to call WebSockerConstructor without subprotocols, like this
```
new WebSocketConstructor(url);
```